_posts/2022-06-01-snitching-on-your-colleagues-using-cloud-magic.md

```diff
@@ -0,0 +1,332 @@
+  * [Before We Dive In](#before-we-dive-in)
+  * [Basic Tasks](#basic-tasks)
+    * [Maven Test](#maven-test)
+    * [Maven Build](#maven-build)
+    * [Copying And Creating An Artifact](#copying-and-creating-an-artifact)
+  * [Containerization](#containerization)
+    * [Fetching Our Artifact](#fetching-our-artifact)
+    * [Creating An Image](#creating-an-image)
+  * [Setting Up A Cloud Database](#setting-up-a-cloud-database)
+    * [Terraform Install](#installing-terraform)
+    * [Terraform Init](#initializing-terraform)
+    * [Terraform Plan](#terraform-plan)
+    * [Terraform Apply](#terraform-apply)
+  * [Provisioning A Cloud Cluster](#provisioning-a-cloud-cluster)
+    * [HELM](#helm-deploy)
+  * [Giving A Signal](#giving-a-signal)
+    * [Creating Our Bot](#creating-our-bot)
+    * [Sending Out Notifications](#sending-out-notifications)
+  * [A Visual Representation](#a-visual-representation)
+* [Out Of The Frying Pan And Into The Fire: Creating Our Own Task](#out-of-the-frying-pan-and-into-the-fire-creating-our-own-task)
+  * [Requirements](#requirements)
+  * [Getting The Logs](#getting-the-logs)
+  * [Filtering And Saving Failures](#filtering-and-saving-failures)
+  * [Factory Fresh](#factory-fresh)
+* [The Berry On Top: Our Physical Feedback](#the-berry-on-top--our-physical-feedback)
+  * [Dollar Store Google Assistant](#dollar-store-google-assistant)
+* [The Good, The Bad & ... The Ugly?](#the-good-the-bad---the-ugly--summary)
+  
+
+# Introduction
+
+7 weeks ago I started my internship at Ordina Mechelen.
+I had several project options available all looking to touch new and unknown tech that might be relevant for future operations.
+
+My inner Judas spoke to me when I saw a listing about a project that would shoot rockets at developers if they broke a build, and it would provide a great opportunity to pull myself out of my comfort zone focusing more on Devops and Cloud platforms rather than pure programming.
+Sadly due to a global hardware shortage the rocket launcher was not available for delivery anymore, so I decided to use a raspberry pi to fetch the build logs and convert them into audio using google text to speech.
+
+
+# The Task At Hand
+
+The project described a ci-cd pipeline that would trigger a raspberry pi once a build fails, then in response the pi would operate a rocket launcher unit that would target the developer responsible for breaking the build.
+As such I broke the final project can be broken down into these steps:
+
+- Create a sacrificial bare-bones spring boot project to put through the ringer
+- Create a pipeline which performs some cookie cutter tasks
+- Expand pipeline to setup AWS infrastructure
+- Expand pipeline to provision said infrastructure
+- Create a custom pipeline task to fetch and push build logs
+- Configure raspberry pi to host a node.js server
+- Listen for new log files
+- If new log files have been found, search for error messages and convert those via TTS
```

Comment:
vermijd afkorting of zet eerst de volledige bewoording met de afkorting erachter om later te hergebruiken. 
Zoals 'text to speech (TTS) '